src/main.rs

```diff
@@ -401,36 +401,31 @@ fn run() -> Result<(), failure::Error> {
     if let Some(matches) = matches.subcommand_matches("config") {
         let api_key = matches.is_present("api-key");
 
-        let mut user = GlobalUser {
-            email: None,
-            api_key: None,
-            api_token: None,
-        };
+        let mut user: GlobalUser;
 
         if !api_key {
             // Default: use API token.
             message::info("Looking to use a Global API Key and email instead? Run \"wrangler config --api-key\". (Not Recommended)");
             println!("Enter API token: ");
             let mut api_token_str: String = read!("{}\n");
             api_token_str.truncate(api_token_str.trim_end().len());
-            if !api_token_str.is_empty() {
-                user.api_token = Some(api_token_str);
-            }
+            user = GlobalUser::TokenAuthUser {
+                api_token: api_token_str,
+            };
         } else {
             message::warn("We don't recommend using your Global API Key! Please consider using an API Token instead. https://support.cloudflare.com/hc/en-us/articles/200167836-Managing-API-Tokens-and-Keys");
             println!("Enter email: ");
             let mut email_str: String = read!("{}\n");
             email_str.truncate(email_str.trim_end().len());
-            if !email_str.is_empty() {
-                user.email = Some(email_str);
-            }
 
             println!("Enter global API key: ");
             let mut api_key_str: String = read!("{}\n");
             api_key_str.truncate(api_key_str.trim_end().len());
-            if !api_key_str.is_empty() {
-                user.api_key = Some(api_key_str);
-            }
+
+            user = GlobalUser::KeyAuthUser {
+                email: email_str,
+                api_key: api_key_str,
```

Comment:
same nit regarding `_str`

Reply:
i'd even go so far as to name the thing that is `api_key` now to `api_key_flag` or `use_api_key`, something that indicates that it is a boolean flag input